### PR TITLE
fix: Domain binding does not take effect

### PR DESCRIPTION
**Pull request by longjuan**
Fixes https://github.com/halo-dev/halo/issues/2846
Add properties domain judgment to externalLink in buildAttachment method.

### CODE REVIEW

File: src/main/java/run/halo/alioss/AliOssProperties.java [hunk 1]
```diff
@@ -33,4 +33,13 @@ public String getObjectName(String filename) {
     enum Protocol {
         http, https
     }
+
+    public void setDomain(String domain) {
+        if (domain.toLowerCase().startsWith("http://")){
```
Comment (JohnNiang) [inline src/main/java/run/halo/alioss/AliOssProperties.java:38]: 这里可能没有考虑到 domain = null 的情况。

Comment (longjuan) [reply src/main/java/run/halo/alioss/AliOssProperties.java]: 确实，https://github.com/halo-sigs/plugin-alioss/pull/7/commits/7e975b0cb2822e560ef92fa4f61a06d596cc2d17

**Review by JohnNiang — APPROVED** [review]
/approve

Thank @longjuan for your first contribution!

**Review by guqing — COMMENTED** [review]
/lgtm
经测试符合预期 🎉

**Review by JohnNiang — APPROVED** [review]
/lgtm

**Pull request merged**